tests/phpunit/tests/post/query.php

```diff
@@ -777,4 +777,525 @@ public function test_found_posts_should_be_integer_even_if_found_posts_filter_re
 
 		$this->assertIsInt( $q->found_posts );
 	}
+
+	/**
+	 * @ticket 47280
+	 */
+	public function test_found_posts_are_correct_for_empty_query() {
+		self::factory()->post->create_many( 12 );
+
+		$q = new WP_Query();
+
+		$this->assertSame( 0, $q->post_count );
+		$this->assertSame( 0, $q->found_posts );
+		$this->assertSame( 0, $q->max_num_pages );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_query_with_no_results( $fields ) {
+		$q = new WP_Query(
+			array(
+				'fields'       => $fields,
+				'posts_status' => 'draft',
+			)
+		);
+
+		$this->assertSame( 0, $q->post_count, self::get_count_message( $q ) );
+		$this->assertSame( 0, $q->found_posts, self::get_count_message( $q ) );
+		$this->assertSame( 0, $q->max_num_pages, self::get_count_message( $q ) );
+	}
+
+	/**
+	 * @ticket 47280
+	 * @dataProvider data_fields
+	 *
+	 * @param string $fields Value of the `fields` argument for `WP_Query`.
+	 */
+	public function test_found_posts_are_correct_for_basic_query( $fields ) {
+		self::factory()->post->create_many( 5 );
```

Comment:
This appears multiple times, is it possible to create it as a shared fixture. It might need to be a CPT to avoid messing up the assertions elsewhere in this file.